src/snapshot.js

```diff
@@ -188,6 +193,10 @@ export function restorePlayerSnapshot(player, snapshotObject) {
   // restoration is required
 
   if (player.ads.videoElementRecycled()) {
+    // Snapshot restore will result in an adended event. That's when
+    // we're really done.
+    player.one('adended', callback);
```

Comment:
are there going to be multiple adended events in some scenarios? One from here and one from redispatch?

Reply:
yeah... normally you'd want the one that happens during content restoration to be `contentended` but that name was already used for something else. I figured calling it adended would get it out of the way. Ideally `contentended` would become `readyforpostroll` and then we could use `contentended` for this to match everything else. Don't think this is the time to do that though.

Reply:
We could also make a new event like `restoreended` if we want. IDK if that's any better.